bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/Ffmpeg.java

```diff
@@ -0,0 +1,205 @@
+                                    countOfMotions--;
+                                    countOfMotions--;
+                                    if (countOfMotions <= 0) {
+                                        ipCameraHandler.noMotionDetected(CHANNEL_FFMPEG_MOTION_ALARM);
+                                    }
+                                }
+                            } else if (line.contains("silence_start")) {
+                                ipCameraHandler.noAudioDetected();
+                            } else if (line.contains("silence_end")) {
+                                ipCameraHandler.audioDetected();
+                            }
+                        } else {
+                            logger.debug("{}", line);
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                logger.warn("An error occured trying to process the messages from FFmpeg.");
+            } finally {
+                switch (format) {
+                    case GIF:
+                        try {
+                            // Without a small delay, Pushover sends no file 10% of time.
+                            Thread.sleep(800);
+                        } catch (InterruptedException e) {
+                        }
+                        logger.debug("Animated GIF has been created and is ready for use.");
+                        ipCameraHandler.setChannelState(CHANNEL_UPDATE_GIF, OnOffType.OFF);
+                        ipCameraHandler.setChannelState(CHANNEL_GIF_HISTORY_LENGTH,
+                                new DecimalType(++ipCameraHandler.gifHistoryLength));
+                        break;
+                    case RECORD:
+                        try {
+                            Thread.sleep(800);
+                        } catch (InterruptedException e) {
+                        }
+                        logger.debug("MP4 has been created and is ready for use.");
+                        ipCameraHandler.setChannelState(CHANNEL_RECORD_MP4, DecimalType.ZERO);
+                        ipCameraHandler.setChannelState(CHANNEL_MP4_HISTORY_LENGTH,
+                                new DecimalType(++ipCameraHandler.mp4HistoryLength));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+    }
+
+    public void startConverting() {
+        if (!streamRunning.isAlive()) {
+            streamRunning = new StreamRunning();
+            logger.debug("Starting ffmpeg with this command now:{}", ffmpegCommand);
+            streamRunning.start();
+            running = true;
+            if (format.equals(ffmpegFormat.HLS)) {
+                ipCameraHandler.setChannelState(CHANNEL_START_STREAM, OnOffType.ON);
+                if (keepAlive > -1) {
+                    try {
+                        Thread.sleep(4500); // Used for on demand HLS to give ffmpeg time to produce the files needed.
```

Comment:
Thanks for the info, helps me learn the why and what to be careful of in future. I have made a change and it would be good to see if this newer method is correct.